src/ripple/ledger/impl/View.cpp

```diff
@@ -1718,6 +1718,14 @@ rippleMPTCredit(
     STAmount saAmount,
     beast::Journal j)
 {
+    if (auto const ter = requireAuth(view, saAmount.issue(), uSenderID);
```

Comment:
I'm not sure if this check should be done here. If we draw a parallel with `rippleCredit` then the check doesn't belong here. How about we do the check in the Payment transactor right before the call to `rippleMPTCredit`? I think that once we add DEX integration, the direct payment is going to be moved to the `MPTEndpointStep` (similar to `XRPEndpointStep`) and the authorization check will be done in the step.